sound/soc/codecs/tfa989x.c

```diff
@@ -80,6 +90,42 @@ static int tfa989x_hw_params(struct snd_pcm_substream *substream,
 					     sr << TFA989X_I2SREG_I2SSR);
 }
 
+static int tfa989x_put_mode(struct snd_kcontrol *kcontrol, struct snd_ctl_elem_value *ucontrol)
+{
+	struct snd_soc_component *component = snd_soc_kcontrol_component(kcontrol);
+	struct tfa989x *tfa989x = snd_soc_component_get_drvdata(component);
+
+	gpiod_set_value_cansleep(tfa989x->rcv_gpiod, ucontrol->value.enumerated.item[0]);
+
+	return snd_soc_put_enum_double(kcontrol, ucontrol);
+}
+
+static const char *mode_text[] = {
+	"Speaker", "Receiver",
+};
+static SOC_ENUM_SINGLE_DECL(mode_enum, TFA989X_I2SREG, TFA989X_I2SREG_RCV, mode_text);
+
+static const struct snd_kcontrol_new tfa9897_controls[] = {
```

Comment:
```suggestion
static const struct snd_kcontrol_new tfa989x_mode_controls[] = {
```

or something like this, since potentially other TFA*s in the future could have this as well